File: examples/java/kora-java-petclinic/src/main/java/io/koraframework/example/petclinic/repository/PetRepository.java
package io.koraframework.example.petclinic.repository;

import io.koraframework.database.common.UpdateCount;
import io.koraframework.database.common.annotation.Id;
import io.koraframework.database.common.annotation.Query;
import io.koraframework.database.common.annotation.Repository;
import io.koraframework.database.jdbc.JdbcRepository;
import io.koraframework.example.petclinic.model.dao.Pet;
import java.util.List;
import java.util.Optional;

@Repository
public interface PetRepository extends JdbcRepository {

    @Query("SELECT %{return#selects} FROM %{return#table} WHERE id = :id")
    Optional<Pet> findById(long id);

    @Query("SELECT %{return#selects} FROM %{return#table} WHERE owner_id = :ownerId ORDER BY name")
    List<Pet> findByOwnerId(long ownerId);

    @Id
    @Query("INSERT INTO %{entity#inserts -= id}")
    long insert(Pet entity);

    @Query("UPDATE %{entity#table} SET %{entity#updates} WHERE %{entity#where = @id}")
    UpdateCount update(Pet entity);
}

File: examples/kotlin/kora-kotlin-petclinic/src/main/kotlin/io/koraframework/kotlin/example/petclinic/repository/PetRepository.kt
package io.koraframework.kotlin.example.petclinic.repository

import io.koraframework.database.common.UpdateCount
import io.koraframework.database.common.annotation.Id
import io.koraframework.database.common.annotation.Query
import io.koraframework.database.common.annotation.Repository
import io.koraframework.database.jdbc.JdbcRepository
import io.koraframework.kotlin.example.petclinic.model.Pet

@Repository
interface PetRepository : JdbcRepository {
    @Query("SELECT %{return#selects} FROM %{return#table} WHERE id = :id")
    fun findById(id: Long): Pet?

    @Query("SELECT %{return#selects} FROM %{return#table} WHERE owner_id = :ownerId ORDER BY name")
    fun findByOwnerId(ownerId: Long): List<Pet>

    @Id
    @Query("INSERT INTO %{entity#inserts -= id}")
    fun insert(entity: Pet): Long

    @Query("UPDATE %{entity#table} SET %{entity#updates} WHERE %{entity#where = @id}")
    fun update(entity: Pet): UpdateCount
}
